task5/index.js

```diff
@@ -0,0 +1,27 @@
+const ul = document.getElementById("js-ul");
+
+const fragment = document.createDocumentFragment();
+const obj = [
+    {to: "bookmark.html", img: "1.png", alt:"画像1", text: "ブックマーク"}, 
+    {to: "message.html", img: "2.png", alt:"画像2", text: "メッセージ"}
+]
+
+const getData = new Promise((resolve) => {
+    resolve(obj);
+});
+
+getData.then(() => {
+    obj.forEach((value) => {
+        const li = document.createElement('li');
+        const a = document.createElement('a');
+        const img = document.createElement('img');
+
+        a.textContent = value.text;
+        a.href = value.to;
+        img.src = value.img;
+        img.alt = value.alt;
+
+        fragment.appendChild(li).appendChild(a).insertBefore(img, a.firstChild);
+    }) 
+    ul.appendChild(fragment);
+});
```

Comment:
すみません、朝お伝えし忘れました…要領悪くて申し訳ありません。
以下ご確認をお願いします。
```suggestion
const getData = new Promise((resolve) => {
    //ココで受け取った解決した値を
    resolve(obj);
});

//ここにvalueとして渡すことができます
getData.then((value) => {
//ここの行で
//console.log(value)すると
//[ 
//{to: "bookmark.html", img: "1.png", alt:"画像1", text: "ブックマーク"},
//{to: "message.html", img: "2.png", alt:"画像2", text: "メッセージ"}
//]
//と出てきます。見てみてください

    value.forEach((item) => {
        const li = document.createElement('li');
        const a = document.createElement('a');
        const img = document.createElement('img');

        a.textContent = item.text;
        a.href = item.to;
        img.src = item.img;
        img.alt = item.alt;

        fragment.appendChild(li).appendChild(a).insertBefore(img, a.firstChild);
    })
    ul.appendChild(fragment);
});
```

Reply:
”利用する”ということがとても勉強になりました。
解決した値は変数を書かずにvalueで表示することができるのですね！

Reply:
> 解決した値は変数を書かずにvalueで表示することができる

ここ必ずvalueって書かなきゃいけないわけじゃないです。今回は仮に分かりやすくvalueと書きました。(伝わりますかね…)

ここに貰ってきた値名を書くと(今回のvalueの事です)受け取った解決された値を使うことができるんです。

なので、下のコメントでも先生が仰ってますが、valueを貰ってきててもobj使っちゃうと意味無いんです😭💦

もう一度ご確認お願い致します…！

Reply:
あんなにわかりやすく解説頂いたのに、修正ミスしてしまい申し訳ないです。
obj使っちゃうとせっかく取得したのに活用できず意味がなくなっちゃうことがよくわかりました。何度もありがとうございます！